zerver/management/commands/send_password_reset_email.py

```diff
@@ -1,46 +1,44 @@
 from __future__ import absolute_import
 
 import logging
-from typing import Any, Dict, List, Optional, Text
+from typing import Any, List, Optional, Text
 
 from argparse import ArgumentParser
-from django.core.management.base import BaseCommand, CommandError
-from django.conf import settings
-from zerver.forms import PasswordResetForm
-from zerver.models import UserProfile, get_user_profile_by_email, get_realm
+from zerver.models import UserProfile
 from django.template import loader
 from django.utils.http import urlsafe_base64_encode
 from django.utils.encoding import force_bytes
 
 from django.contrib.auth.tokens import default_token_generator, PasswordResetTokenGenerator
 
 from zerver.lib.send_email import send_email
+from zerver.lib.management import ZulipBaseCommand
 
-class Command(BaseCommand):
+class Command(ZulipBaseCommand):
     help = """Send email to specified email address."""
 
     def add_arguments(self, parser):
         # type: (ArgumentParser) -> None
         parser.add_argument('--to', metavar='<to>', type=str,
                             help="email of user to send the email")
-        parser.add_argument('--realm', metavar='<realm>', type=str,
-                            help="realm to send the email to all users in")
         parser.add_argument('--server', metavar='<server>', type=str,
                             help="If you specify 'YES' will send to everyone on server")
+        self.add_realm_args(parser)
 
     def handle(self, *args, **options):
         # type: (*Any, **str) -> None
-        if options["to"]:
-            users = [get_user_profile_by_email(options["to"])]
-        elif options["realm"]:
-            realm = get_realm(options["realm"])
+        realm = self.get_realm(options)
+        if realm is not None and options["to"] is None:
             users = UserProfile.objects.filter(realm=realm, is_active=True, is_bot=False,
                                                is_mirror_dummy=False)
+        elif options["to"]:
+            users = [self.get_user(options["to"], realm)]
         elif options["server"] == "YES":
             users = UserProfile.objects.filter(is_active=True, is_bot=False,
                                                is_mirror_dummy=False)
         else:
-            raise RuntimeError("Missing arguments")
+            self.print_help("./manage.py", "send_password_reset_email")
+            exit(1)
```

Comment:
I think this one we want to refactor a bit more, since we're changing the help for `--realm` pretty significantly.  My thought is maybe this should now have a `--target=realm` and `--target=server` for you to specify if you want to send to an entire realm or server (basically replacing the `--server` argument).  That seems like a cleaner solution that won't be potentially confusing.